### ISSUE TITLE
Release checklist

### ISSUE THREAD

**Issue by ericfranz** [OSC/frame-renderer#14]
- [x] Add configuration singleton with clusters and dataroot (and the other minimal configs as required)
   - https://github.com/OSC/ood-myjobs/blob/3437bfd2e4be121637890af485a2b2b385053ec4/config/configuration_singleton.rb (the ConfigurationSingleton singleton offers two benefits: 1. we end up with a single global object `Configuration` with methods to call to return configuration, which can return set values or default values, or have some complex logic to determine at request time what to return (as opposed to [setting values on a `Rails.configuration.x` object at initialization time](https://guides.rubyonrails.org/v4.2/configuring.html#custom-configuration)). The bigger benefit is we can [easily write readable automated tests](https://github.com/OSC/ood-dashboard/blob/ac94fc2d34efaa34a249b3bde7eeb20b53b8daa3/test/system/configuration_singleton_test.rb) against `ConfigurationSingleton` without having to worry about the state of some global object
   - https://github.com/OSC/ood-myjobs/blob/3437bfd2e4be121637890af485a2b2b385053ec4/config/boot.rb#L8-L13 (adding to boot allows using `Configuration.xyz` in bin/setup in config/application.rb etc.)
- [x] Add the [bin/setup-production script](https://github.com/OSC/ood-myjobs/blob/master/bin/setup-production) - FYI this is a poorly named script probably should be `bin/per-user-setup` but alas.
- [x] Update [bin/setup](https://github.com/OSC/ood-myjobs/blob/3437bfd2e4be121637890af485a2b2b385053ec4/bin/setup) to work like the others, handling the desired default RAILS_RELATIVE_URL_ROOT etc.
- [x] Add the json attrs fields to models as escape hatch for future work
   - https://github.com/OSC/ood-myjobs/blob/3437bfd2e4be121637890af485a2b2b385053ec4/app/models/workflow.rb#L8-L11 - lets us add more form elements and attributes to a model without requiring a database migration
   - https://github.com/OSC/ood-myjobs/blob/3437bfd2e4be121637890af485a2b2b385053ec4/db/schema.rb#L38
- [x] Add type column to script (?) and project (?) as escape hatch for future work?
   - Basically [this commit](https://github.com/OSC/ood-myjobs/commit/33b4cc14f629f1debb14f5b899c0fd5aa26a54f1) which arguably should be in all of our Rails apps allowing us to create new models and attributes without the need for any new migrations
- [x] RPM (#12)
- [x] Test this app's job template vs the original maya app to confirm no significant difference, and revert to original maya job template (using job arrays) as necessary

**Comment by johrstrom** [comment]
I'm not sure how `Add the json attrs fields` and `Add type column to script (?) and project (?)` are different? So that's why I've left the one item blank, I've not implemented it yet.

**Comment by ericfranz** [comment]
- json attrs is the new column for json store so you can add arbitrary attributes to existing models
- the `JsonStore` model with a type column so you can add new models using single table inheritance
- other two things that need fixed:
  - [ ] manifest category should be "Gateway Apps" and subcategory should be "Graphics" (well I don't know what makes the most sense for the subcategory)
  - [ ] puppet needs to be updated for that apps so that the permissions are 750 and the group that owns the app is the maya group

**Comment by ericfranz** [comment]
Note: The only things that matter for release right now are the manifest and puppet fix. I think the other two can be delivered in whatever state they are in. It will not be catastrophic if we deploy a new version of the app with a schema change.
